snmplib/snmp_api.c

```diff
@@ -4378,6 +4379,11 @@ _snmp_parse(struct session_list *slp,
                     pdu->securityName, secLevelName[pdu->securityLevel],
                     snmp_api_errstring(result)));
 
+        if (result == SNMPERR_USM_UNKNOWNSECURITYNAME) {
+            snmp_call_callbacks(SNMP_CALLBACK_APPLICATION,
+                                SNMPD_CALLBACK_AUTH_FAILURE, (void *)pdu);
```

Comment:
Is this cast necessary?

Reply:
Sorry, no, it is not necessary (it was based on a similar callback call, but it appears the cast has been removed there as well).